doc/about/customisation/index.md

```diff
@@ -0,0 +1,48 @@
+# Customisation {#about_customisation}
+
+SeqAn clearly documents which of its interfaces are designed to be used with user-provided types and how
+this can be realised.
+
+[TOC]
+
+# Generic interfaces
+
+Templates are a core part of generic programming in C++, because they allow instantiating the code with different
+types or constants. SeqAn's API consists mostly of templates for exactly this reason, although we try to make this
+less visible by having the template parameters deduced in many scenarios.
+
+We strive to document the exact requirements for every template parameter in the API documentation; often these
+are also enforced via C++ concepts. Wherever other types meet the given requirements, you can use these
+type as arguments – independent of whether they are standard library types, your own types or types from another
+library. See \ref tutorial_concepts for more details on this.
+
+Some requirements are more elaborate and depend on a specialisation or "overload" of another entity from the
+SeqAn library. We call these "customisation points" (also known as "extension points").
+
+# Customisation points
+
+One such customisation point is seqan3::to_rank. Instead of the seqan3::Semialphabet concept looking directly for
+member functions or free functions in you namespace, it always checks for a valid definition of seqan3::to_rank which
+in turn resolves to different possible implementations.
+
+The design pattern we use for customisation points is often called a "Niebloid" (after Eric Niebler); more technical
```

Comment:
I wouldn't put this in here. These are officially called Customisation Point Objects and I would stick to it. Many people won't know what a Niebloid really is.